packages/m2m-gateway/src/services/purposeService.ts

```diff
@@ -122,5 +140,34 @@ export function purposeServiceBuilder(clients: PagoPAInteropBeClients) {
 
       return toM2mGatewayApiPurposeVersion(version);
     },
+    createPurposeVersion: async (
+      purposeId: PurposeId,
+      versionSeed: m2mGatewayApi.PurposeVersionSeed,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.PurposeVersion> => {
+      logger.info(`Creating purpose version`);
+
+      const versionResponse =
+        await clients.purposeProcessClient.createPurposeVersion(versionSeed, {
```

Comment:
This can also passthrough a 409 Conflict which is missing in the M2M API spec for this call, I think